sklearn/linear_model/tests/test_quantile.py

```diff
@@ -0,0 +1,126 @@
+    if coef is not None:
+        assert_allclose(model.coef_[0], coef, atol=1e-2)
+    if alpha < 100:
+        assert model.coef_[0] >= 1
+    assert model.coef_[0] <= 10
+
+
+@pytest.mark.skipif(sp_version < parse_version("1.0.0"), reason=_SCIPY_TOO_OLD)
+def test_quantile_equals_huber_for_low_epsilon():
+    X, y = make_regression(n_samples=100, n_features=20, random_state=0,
+                           noise=1.0)
+    huber = HuberRegressor(epsilon=1+1e-4, alpha=1e-4).fit(X, y)
+    quant = QuantileRegressor(alpha=1e-4).fit(X, y)
+    assert_allclose(huber.intercept_, quant.intercept_, atol=1e-1)
+    assert_allclose(huber.coef_, quant.coef_, atol=1e-1)
+
+
+@pytest.mark.skipif(sp_version < parse_version("1.0.0"), reason=_SCIPY_TOO_OLD)
+def test_quantile_estimates_fraction():
+    # Test that model estimates percentage of points below the prediction
+    X, y = make_regression(n_samples=1000, n_features=20, random_state=0,
+                           noise=1.0)
+    for q in [0.5, 0.9, 0.05]:
+        quant = QuantileRegressor(quantile=q, alpha=0).fit(X, y)
+        fraction_below = np.mean(y < quant.predict(X))
+        assert_allclose(fraction_below, q, atol=1e-2)
+
+
+@pytest.mark.skipif(sp_version < parse_version("1.0.0"), reason=_SCIPY_TOO_OLD)
+def test_quantile_without_intercept():
+    X, y = make_regression(n_samples=300, n_features=20, random_state=0,
+                           noise=1.0)
+    quant = QuantileRegressor(alpha=1e-4, fit_intercept=False).fit(X, y)
+    # check that result is similar to Huber
+    huber = HuberRegressor(epsilon=1 + 1e-4, alpha=1e-4, fit_intercept=False
+                           ).fit(X, y)
+    assert_allclose(huber.intercept_, quant.intercept_, atol=1e-1)
+    assert_allclose(huber.coef_, quant.coef_, atol=1e-1)
+    # check that we still predict fraction
+    fraction_below = np.mean(y < quant.predict(X))
+    assert_allclose(fraction_below, 0.5, atol=1e-1)
+
+
+@pytest.mark.skipif(sp_version < parse_version("1.0.0"), reason=_SCIPY_TOO_OLD)
+def test_quantile_sample_weight():
+    # test that with unequal sample weights we still estimate weighted fraction
+    n = 1000
+    X, y = make_regression(n_samples=n, n_features=10, random_state=0,
+                           noise=10.0)
+    weight = np.ones(n)
+    # when we increase weight of upper observaions,
+    # estimate of quantile should go up
+    weight[y > y.mean()] = 100
+    quant = QuantileRegressor(quantile=0.5, alpha=1e-4)
+    quant.fit(X, y, sample_weight=weight)
+    fraction_below = np.mean(y < quant.predict(X))
+    assert fraction_below > 0.5
+    weighted_fraction_below = np.sum((y < quant.predict(X)) * weight) \
+        / np.sum(weight)
+    assert_allclose(weighted_fraction_below, 0.5, atol=1e-2)
+
+
+@pytest.mark.skipif(sp_version < parse_version("1.0.0"), reason=_SCIPY_TOO_OLD)
+@pytest.mark.parametrize('quantile', [2.0, 1.0, 0.0, -1])
+def test_quantile_incorrect_quantile(quantile):
```

Comment:
Could you place this as the first test in the file and also test for the other parameters like `alpha`, `fit_intercept`, ....
The following link might help:
https://github.com/scikit-learn/scikit-learn/blob/114616d9f6ce9eba7c1aacd3d4a254f868010e25/sklearn/ensemble/_hist_gradient_boosting/tests/test_gradient_boosting.py#L47-L68